sdk/eventhub/event-hubs/src/checkpointManager.ts

```diff
@@ -1,58 +1,94 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-// import { PartitionContext } from "./partitionContext";
-import { EventData } from "./eventData";
-// import { PartitionManager } from "./eventProcessor";
+import { PartitionContext } from "./partitionContext";
+import { ReceivedEventData } from "./eventData";
+import { PartitionManager } from "./eventProcessor";
+import { generate_uuid } from "rhea-promise";
 
 /**
  * Used by createCheckpoint in PartitionManager
  **/
 export interface Checkpoint {
-   /**
+  /**
    * @property The event hub name
    */
   eventHubName: string;
-   /**
+  /**
    * @property The consumer group name
    */
   consumerGroupName: string;
-   /**
+  /**
    * @property The unique instance identifier
    */
   instanceId: string;
-   /**
+  /**
    * @property The identifier of the Event Hub partition
    */
   partitionId: string;
-   /**
+  /**
    * @property The sequence number of the event.
    */
   sequenceNumber: number;
-   /**
+  /**
    * @property The offset of the event.
    */
   offset: number;
+  /**
+   * @property The unique identifier for the operation.
+   */
+  eTag: string;
 }
 
 /**
  * CheckPointManager is created by the library & passed to user's code to let them create a checkpoint
  */
 export class CheckpointManager {
-  // private _partitionContext: PartitionContext; // for internal use by createCheckpoint
-  // private _partitionManager: PartitionManager; // for internal use by createCheckpoint
-
-  // constructor(partitionContext: PartitionContext, partitionManager: PartitionManager) {
-  //   this._partitionContext = partitionContext;
-  //   this._partitionManager = partitionManager;
-  // }
-
-  public async updateCheckpoint(eventData: EventData): Promise<void>;
+  private _partitionContext: PartitionContext; // for internal use by createCheckpoint
+  private _partitionManager: PartitionManager; // for internal use by createCheckpoint
+  private _instanceId: string;
+  private _eTag: string;
 
-  public async updateCheckpoint(offset: string, sequenceNumber: number): Promise<void>;
+  constructor(partitionContext: PartitionContext, partitionManager: PartitionManager) {
+    this._partitionContext = partitionContext;
+    this._partitionManager = partitionManager;
+    this._eTag = "";
+    this._instanceId = generate_uuid();
```

Comment:
I thought this instance id was the id for the Event Processor.
If we generate a new one for each checkpoint manager, then we will get a new id for each partition